pyomo/contrib/alternative_solutions/solution.py

```diff
@@ -9,150 +9,285 @@
-    get_objective_name_values(self):
-        Get a dictionary of objective name-objective value pairs.
+    value : float
+        The value of the variable.
+    fixed : bool
+        If True, then the variable was fixed during optimization.
+    name : str
+        The name of the variable.
+    index : int
+        The unique identifier for this variable.
+    discrete : bool
+        If True, then this is a discrete variable
+    suffix : dict
+        Other information about this variable.
     """
 
-    def __init__(self, model, variable_list, include_fixed=True, objective=None):
-        """
-        Constructs a Pyomo Solution object.
+    value: float = nan
+    fixed: bool = False
+    name: str = None
+    repn = None
+    index: int = None
+    discrete: bool = False
+    suffix: MyMunch = dataclasses.field(default_factory=MyMunch)
+
+    def to_dict(self):
+        return dataclasses.asdict(self, dict_factory=_custom_dict_factory)
+
+
+@dataclasses.dataclass(**dataclass_kwargs)
+class ObjectiveInfo:
+    """
+    Represents an objective in a solution.
+
+    Attributes
+    ----------
+    value : float
+        The objective value.
+    name : str
+        The name of the objective.
+    index : int
+        The unique identifier for this objective.
+    suffix : dict
+        Other information about this objective.
+    """
+
+    value: float = nan
+    name: str = None
+    index: int = None
+    suffix: MyMunch = dataclasses.field(default_factory=MyMunch)
+
+    def to_dict(self):
+        return dataclasses.asdict(self, dict_factory=_custom_dict_factory)
+
+
+@functools.total_ordering
+class Solution:
+    """
+    An object that describes an optimization solution.
+
+    Parameters
+    -----------
+    variables : None or list
+        A list of :py:class:`VariableInfo` objects. (default is None)
+    objective : None or :py:class:`ObjectiveInfo`
+        A :py:class:`ObjectiveInfo` object. (default is None)
+    objectives : None or list
+        A list of :py:class:`ObjectiveInfo` objects. (default is None)
+    kwds : dict
+        A dictionary of auxiliary data that is stored with the core solution values.
+    """
+
+    def __init__(self, *, variables=None, objective=None, objectives=None, **kwds):
+        self.id = None
+
+        self._variables = []
+        self.name_to_variable = {}
+        self.fixed_variable_names = set()
+        if variables is not None:
+            self._variables = variables
+            for v in variables:
+                if v.name is not None:
+                    if v.fixed:
+                        self.fixed_variable_names.add(v.name)
+                    self.name_to_variable[v.name] = v
+
+        self._objectives = []
+        self.name_to_objective = {}
+        if objective is not None:
+            objectives = [objective]
+        if objectives is not None:
+            self._objectives = objectives
+            for o in objectives:
+                if o.name is not None:
```

Comment:
Same question here.

Reply:
Same as above. I want to make it convenient to specify objective data in a manner that best suits their application: single-objective of multi-objective.